internal/impl/snowflake/streaming/int128/division.go

```diff
@@ -58,6 +58,28 @@ func Div(dividend, divisor Num) Num {
 	return quotient
 }
 
+// Compare returns -1 if a < b, 0 if a == b, and 1 if a > b.
+func Compare(a, b Num) int {
+	if a.hi < b.hi || (a.hi == b.hi && a.lo < b.lo) {
+		return -1
+	}
+	if a.hi == b.hi && a.lo == b.lo {
+		return 0
+	}
+	return 1
+}
+
+// CompareUnsigned returns -1 if |a| < |b|, 0 if a == b, and 1 if |a| > |b|.
+func CompareUnsigned(a, b Num) int {
+	if uint64(a.hi) < uint64(b.hi) || (a.hi == b.hi && a.lo < b.lo) {
+		return -1
+	}
+	if a.hi == b.hi && a.lo == b.lo {
+		return 0
+	}
+	return 1
+}
```

Comment:
I wonder if we can leverage https://pkg.go.dev/cmp#Compare for these. Not sure if that's possible

Reply:
Yes we can make this much more readable :) Thanks for this!